seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java

```diff
@@ -129,9 +131,45 @@ public void memberRemoved(MembershipServiceEvent event) {
 
     @Override
     public CompletableFuture<List<SlotProfile>> applyResources(
-            long jobId, List<ResourceProfile> resourceProfile) throws NoEnoughResourceException {
+            long jobId, List<ResourceProfile> resourceProfile, Map<String, String> tags)
+            throws NoEnoughResourceException {
         waitingWorkerRegister();
-        return new ResourceRequestHandler(jobId, resourceProfile, registerWorker, this).request();
+        ConcurrentMap<Address, WorkerProfile> matchedWorker;
+        if (tags == null || tags.isEmpty()) {
+            matchedWorker = registerWorker;
+        } else {
+            matchedWorker =
+                    registerWorker.entrySet().stream()
+                            .filter(
+                                    e -> {
+                                        Map<String, String> workerAttr =
+                                                e.getValue().getAttributes();
+                                        if (workerAttr == null || workerAttr.isEmpty()) {
+                                            return true;
+                                        }
```

Comment:
Why workerAttr is empty it will be matched?

Reply:
fixed